x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/rest/action/RestMonitoringBulkAction.java

```diff
@@ -75,7 +76,9 @@ public String getName() {
 
     @Override
     public RestChannelConsumer doPrepareRequest(RestRequest request, XPackClient client) throws IOException {
-        final String defaultType = request.param("type");
+        if (Strings.isEmpty(request.param("type")) == false) {
+            logger.error("Custom types for monitoring is not supported");
```

Comment:
Why are we logging instead of failing with an exception? Are we sure that 6.7 never uses the bulk monitoring API with a type in the path? If not I think we should keep support for the `defaultType` for now and remove it in 8.0. If we are sure that the type is never set via the path, then maybe we should remove the path from the REST handler entirely?

Reply:
I am like 99% sure that Kibana/Beats/LS don't send this parameter. I went ahead and moved to an throwing an exception if it is sent.

Reply:
Should we instead remove the paths that accept a type in the constructor of `RestMonitoringBulkAction`?

Reply:
Ahh, yes (sorry a bit slow here).  I removed this on https://github.com/elastic/elasticsearch/pull/39888/commits/1573aae01296ffbbeb98bca81e0fc1d2214860ad.